### PR TITLE
[Fix #699] `use_local_fs` with slash in branch name

### LINKED ISSUE #699: `use_local_fs` not working for branches with `/`
## Issue Description

Type: *bug report*

### Describe what happened (or what feature you want)

- `use_local_fs` doesn't work for branches with `/`, e.g. `developer/feature`

When starting a review on the branch, it first (incorrectly) reports:
```
Currently not in PR branch, would you like to checkout?
(Y)es, [N]o:
```

Regardless of your choice, the review windows does not use the local fs.

### Describe what you expected to happen

- "Currently not in PR branch" message doesn't show when already on the PR branch
- `use_local_fs` correctly works for branches with `/`

### How to reproduce it (as minimally and precisely as possible)

1. Create PR from branch with `/`
2. Checkout branch
3. `:Octo {PR URL}`
4. `:Octo review start`

### Tell us your environment

- MacOS Sequoia 15.1
- NVIM v0.10.2, Build type: Release, LuaJIT 2.1.1731601260
- Octo `{ "branch": "master", "commit": "14d412f517a3cbea4cd4299207ee067fde542062" }`

### Anything else we need to know?

I believe I've tracked it down to https://github.com/pwntester/octo.nvim/blob/14d412f517a3cbea4cd4299207ee067fde542062/lua/octo/utils.lua#L378-L392, happy to take a look myself when I find some time

### PR DESCRIPTION
**Pull request by nickthegroot**
### Describe what this PR does / why we need it

See issue for full explanation

TL;DR: Using `use_local_fs` for branches with `/` in the name (e.g. `name/feature`) would cause Octo to think that the local branch never matched the PR branch. 

### Does this pull request fix one issue?

Fixes #699 

### Describe how you did it

Instead of taking the first `/` after the remote, we concat all leftover entries in the table.

### Describe how to verify it

1. Create PR from branch with /
1. Checkout branch
1. `:Octo {PR URL}`
1. `:Octo review start`

### Checklist

- [X] Passing tests and linting standards
- [X] Documentation updates in README.md and doc/octo.txt

### CODE REVIEW

**Review by GuillaumeLagrange — APPROVED** [review]
lgtm

I do believe the experience of using local fs can be improved when handling PRs from forks that are not already present as a git remote, but I will create a dedicated issue, since this just fixes something already broken without introducing any regression.

Thanks for your contribution!
